WordPress/gutenberg
# Fix navigation block submenu focus when closing menu with the ESC key

## What?
Fix a bug that causes a wrong focus when closing the submenu using the ESC key.

## Why?
Right now, the focus after closing a submenu is not correct if you previously tab inside the submenu.

## How?

It seems that the `openMenu` function is being called each time we navigate through the submenu items, so it change the `previosFocus` element.

I've changed the event to `focus` instead of `focusin`. According to [its docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event), `focusin` bubbles while `focus` does not. This means that the `openMenu` is called only once when the menu is open.

## Testing Instructions

1. Add a navigation block with submenu items.
2. Navigate through the menu with the keyboard and press ENTER to open the submenu.
3. Navigate through the submenu items pressing tab.
4. Close the menu with the ESC key.

Before this fix, you should see that the focus is lost. After the fix, focus should be on the menu item.

## Screenshots or screencast

https://github.com/WordPress/gutenberg/assets/34552881/37799e64-a6f7-49d6-aa55-0779c1a0b306
